src/foam/dao/MDAO.java

```diff
@@ -107,11 +107,44 @@ public MDAO(ClassInfo of) {
     index_ = new AltIndex(new TreeIndex((PropertyInfo) this.of_.getAxiomByName("id")));
   }
 
+
+
+  public ArrayList<PropertyInfo> propertyInfoList_(Index index){
+        ArrayList<PropertyInfo> infoList = new ArrayList<PropertyInfo>();
+        var current = index;
+        while ( current != null && ( current instanceof TreeIndex ) ){
+          infoList.add(((TreeIndex)current).getProp());
+          current = ((TreeIndex)current).getTail();
+        }
+        return infoList;
+
+  }
+  public boolean isSubSet_(ArrayList<PropertyInfo> list1 , ArrayList<PropertyInfo> list2){
+    return list2.containsAll(list1);
+  }
+
+
+
+
   public void addIndex(Index index) {
     synchronized ( writeLock_ ) {
+      if ( index instanceof TreeIndex ){
+        var properties = propertyInfoList_(index);
+        var delegates = index_.getDelegates();
+        for ( Index i : delegates ){
+          var properties2 = propertyInfoList_(i);
+          if ( isSubSet_(properties,properties2) ){
```

Comment:
Should this be:

if ( isSubSet_(properties,properties2) || isSubSet_(properties2, properties) ){